### PR TITLE
Issue/7209 replace deprecated pm methods

### LINKED ISSUE #7209: Replace deprecated getPackageInfo, resolveService, queryIntentActivities
```
Deprecated
Use getPackageInfo(String, PackageManager.PackageInfoFlags) instead.
```

```
Deprecated
Use resolveService(Intent, PackageManager.ResolveInfoFlags) instead.
```

```
Deprecated
Use queryIntentActivities(Intent, PackageManager.ResolveInfoFlags) instead.
```

### PR DESCRIPTION
**Pull request by codedsun**
Closes: #7209 

### Description

Replace deprecated pm methods

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by codedsun** [comment]
Hey @kidinov - Please review the PR when time allows. Do let me know if any changes are required. Thanks

File: WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt [hunk 1]
```diff
@@ -168,11 +171,19 @@ object AppPrefs {
      */
     val installationDate: Date?
         get() = try {
-            context
-                .packageManager
-                .getPackageInfo(context.packageName, 0)
-                .firstInstallTime
-                .let { Date(it) }
+            if(SystemVersionUtils.isAtLeastT()) {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt]: How do you look into creating a `PackageManagerExt` file with extension functions something like:


```
fun PackageManager.packageInfo(packageName: String, flags: Long): PackageInfo = when {
    SDK_INT >= TIRAMISU -> getPackageInfo(packageName, PackageManager.PackageInfoFlags.of(flags.toLong()))
    else -> @Suppress("DEPRECATION") getPackageInfo(packageName, flags)
}
```

In this case, the only change to the code will be:

```
.getPackageInfo(context.packageName, 0) -> .packageInfo(context.packageName, 0L)
```

A similar approach can be applied for other places

**Review by kidinov — COMMENTED** [review]
@codedsun Huge thanks for working on this 🙏

The change in general looks good. I would just consider:
* Extract it into some [reusable form](https://github.com/woocommerce/woocommerce-android/pull/7237/files#r951007469)
* Run `./gradlew detektAll` to make sure that formatting is correct
* The current change, does not remove the warnings. They should be suppressed

**Comment by codedsun** [comment]
Hey @kidinov , Thanks for the suggestion. Have implemented them, do let me know if any changes are required.

**Comment by kidinov** [comment]
@codedsun Huge thanks for your contribution! I'll take it from here

**Pull request merged**